lightgbm/src/main/scala/com/microsoft/azure/synapse/ml/lightgbm/BasePartitionTask.scala

```diff
@@ -140,6 +140,13 @@ abstract class BasePartitionTask extends Serializable with Logging {
 
       try {
         if (taskCtx.shouldExecuteTraining) {
+          //close socket before lightgbm bind port
+          try {
+            taskCtx.networkTopologyInfo.localSocket.close()
+          } catch {
+            case e: Exception => log.warn("close local bind port socket failed ")
```

Comment:
this looks like its a bit generic of a error, can we make it more specific so we dont catch other errors?